### PR TITLE
Adds semicolon option `ignore-bound-class-methods`

### LINKED ISSUE #1476: Semicolon rule should flag unnecessary semicolons on class member functions
We changed the behavior of the semicolon rule to allow omission of the trailing semicolon after bound class member functions:

``` ts
class Foo {
  public renderFoo = () => {
    return null;
  }
   ^ right here
}
```

I propose that we actually _ban_ semicolons at that location in TSLint 4.0. We can support the old functionality with a new option `allow-bound-class-methods`.

### PR DESCRIPTION
**Pull request by nchen63**
also changes default behavior by requiring semicolons to not come after bound class methods unless this setting is used

closes #1476

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/tslint](https://github.com/palantir/tslint), @nchen63! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by nchen63** [comment]
@JKillian @adidahiya

File: docs/rules/semicolon/index.html [hunk 1]
```diff
@@ -10,6 +10,7 @@
```
Comment (JKillian) [inline docs/rules/semicolon/index.html]: this file update shouldn't be in this PR, it should only happen when you're updating the docs as a whole (on release)

File: src/rules/semicolonRule.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ import * as Lint from "../lint";
 
 const OPTION_ALWAYS = "always";
 const OPTION_NEVER = "never";
+const OPTION_ALLOW_BOUND_CLASS_METHODS = "allow-bound-class-methods";
```
Comment (JKillian) [inline src/rules/semicolonRule.ts]: maybe it should be "skip-bound-class-methods"?

Comment (JKillian) [inline src/rules/semicolonRule.ts]: oh wait, we already have an "ignore-interfaces" option, this should be "ignore-bound-class-methods" then

File: src/rules/semicolonRule.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ import * as Lint from "../lint";
 
 const OPTION_ALWAYS = "always";
 const OPTION_NEVER = "never";
+const OPTION_ALLOW_BOUND_CLASS_METHODS = "ignore-bound-class-methods";
```
Comment (JKillian) [inline src/rules/semicolonRule.ts]: naming of variable

**Pull request merged**